diff --git a/pylib/gyp/xcode_emulation.py b/pylib/gyp/xcode_emulation.py
index dba8e769..8f764ccc 100644
--- a/pylib/gyp/xcode_emulation.py
+++ b/pylib/gyp/xcode_emulation.py
@@ -1403,16 +1403,16 @@ def XcodeVersion():
   except:
     version = CLTVersion()
     if version:
-      version = re.match(r'(\d\.\d\.?\d*)', version).groups()[0]
+      version = re.search(r'^(\d{1,2}\.\d(\.\d+)?)', version).groups()[0]
     else:
       raise GypError("No Xcode or CLT version detected!")
     # The CLT has no build information, so we return an empty string.
     version_list = [version, '']
   version = version_list[0]
   build = version_list[-1]
-  # Be careful to convert "4.2" to "0420":
-  version = version.split()[-1].replace('.', '')
-  version = (version + '0' * (3 - len(version))).zfill(4)
+  # Be careful to convert "4.2" to "0420" and "10.0" to "1000":
+  version = format(''.join((version.split()[-1].split('.') + ['0', '0'])[:3]),
+                   '>04s')
   if build:
     build = build.split()[-1]
   XCODE_VERSION_CACHE = (version, build)
